src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,243 @@
+      }
+
+      private _getEdges(p: Point) {
+        var edges = {
+          top: false,
+          bottom: false,
+          left: false,
+          right: false
+        };
+
+        var bounds = this.bounds();
+        var t = bounds.topLeft.y;
+        var b = bounds.bottomRight.y;
+        var l = bounds.topLeft.x;
+        var r = bounds.bottomRight.x;
+        var rad = this._detectionRadius;
+
+        if (l - rad <= p.x && p.x <= r + rad) {
+          edges.top = (t - rad <= p.y && p.y <= t + rad);
+          edges.bottom = (b - rad <= p.y && p.y <= b + rad);
+        }
+
+        if (t - rad <= p.y && p.y <= b + rad) {
+          edges.left = (l - rad <= p.x && p.x <= l + rad);
+          edges.right = (r - rad <= p.x && p.x <= r + rad);
+        }
+
+        return edges;
+      }
+
+      public _doRender() {
+        super._doRender();
+        if (this.boxVisible()) {
+          var bounds = this.bounds();
+          var t = bounds.topLeft.y;
+          var b = bounds.bottomRight.y;
+          var l = bounds.topLeft.x;
+          var r = bounds.bottomRight.x;
+
+          this._boxEdgeT.attr({
+            x1: l, y1: t, x2: r, y2: t
+          });
+          this._boxEdgeB.attr({
+            x1: l, y1: b, x2: r, y2: b
+          });
+          this._boxEdgeL.attr({
+            x1: l, y1: t, x2: l, y2: b
+          });
+          this._boxEdgeR.attr({
+            x1: r, y1: t, x2: r, y2: b
+          });
+
+          this._boxCornerTL.attr({ cx: l, cy: t });
+          this._boxCornerTR.attr({ cx: r, cy: t });
+          this._boxCornerBL.attr({ cx: l, cy: b });
+          this._boxCornerBR.attr({ cx: r, cy: b });
+        }
+      }
+
+      /**
+       * Gets the detection radius of the drag box.
+       *
+       * @return {number} The detection radius of the drag box.
+       */
+      public detectionRadius(): number;
+      /**
+       * Sets the detection radius of the drag box.
+       *
+       * @param {number} r The desired detection radius.
+       * @return {DragBoxLayer} The calling DragBoxLayer.
+       */
+      public detectionRadius(r: number): DragBoxLayer;
+      public detectionRadius(r?: number): any {
+        if (r == null) {
+          return this._detectionRadius;
+        }
+        if (r < 0) {
+          throw new Error("detection radius cannot be negative.");
+        }
+        this._detectionRadius = r;
+        if (this._isSetup) {
+          this._boxEdgeT.style("stroke-width", this._detectionRadius * 2);
+          this._boxEdgeB.style("stroke-width", this._detectionRadius * 2);
+          this._boxEdgeL.style("stroke-width", this._detectionRadius * 2);
+          this._boxEdgeR.style("stroke-width", this._detectionRadius * 2);
+          this._boxCornerTL.attr("r", this._detectionRadius);
+          this._boxCornerTR.attr("r", this._detectionRadius);
+          this._boxCornerBL.attr("r", this._detectionRadius);
+          this._boxCornerBR.attr("r", this._detectionRadius);
+        }
```

Comment:
This code should probably be moved to a function like
applyDetectionRadius or similar.  As a result of this function doing all the work, there's an awkward this.detectionRadius(this._detectionRadius) up above.

Also, if you want to be ultra safe, you can stick with this._render but that is up to you as this is indeed more efficient